### PR TITLE
Add transient storage slot support in StorageSlot.sol

### LINKED ISSUE #4961: Library for derivation of slots
Solidity does the derivation of slot under the hood when trying to access a value in an array or a mapping. 
Users may want to do te same to compute slots for transient storage, which solidity does not support yet.

[This library](https://github.com/Amxx/EIP-5806/blob/main/packages/test-uniswap/contracts/libs/Slots.sol) does exactly that.

See it used [here](https://github.com/Amxx/EIP-5806/blob/main/packages/test-uniswap/contracts/libs/ERC20TemporaryAllowance.sol#L45-L51)

### PR DESCRIPTION
**Pull request by Amxx**
Replaces #4965
Fixes #4961

~~Blocked by the `hardhat-exposed` issue: [see this](https://github.com/frangio/hardhat-exposed/pull/32)~~

#### PR Checklist

- [x] Tests
- [x] Documentation
- [x] Changeset entry (run `npx changeset add`)

### CODE REVIEW

File: contracts/utils/StorageSlot.sol [hunk 1]
```diff
@@ -29,6 +29,24 @@ pragma solidity ^0.8.20;
  * }
  * ```
  *
+ * Since version 5.1, this library also support writing and reading value types to and from transient storage.
+ *
+ *  * Example usaging transiant storage:
```
Comment (Amxx) [inline contracts/utils/StorageSlot.sol:34]:
```suggestion
 *  * Example usage of transient storage:
```

File: hardhat.config.js [hunk 2]
```diff
@@ -90,19 +96,21 @@ module.exports = {
...
+      hardfork: argv.evm,
       allowUnlimitedContractSize,
       initialBaseFeePerGas: argv.coverage ? 0 : undefined,
     },
   },
   exposed: {
     imports: true,
     initializers: true,
-    exclude: ['vendor/**/*', '**/*WithInit.sol'],
+    exclude: ['vendor/**/*', '**/*WithInit.sol', 'utils/TransientSlot.sol'],
```
Comment (Amxx) [inline hardhat.config.js:113]:
```suggestion
    exclude: ['vendor/**/*', '**/*WithInit.sol'],
```

File: scripts/generate/templates/StorageSlot.js [hunk 3]
```diff
@@ -28,7 +28,25 @@ pragma solidity ^0.8.20;
  *     }
  * }
  * \`\`\`
- * 
+ *
+ * Since version 5.1, this library also support writing and reading value types to and from transient storage.
+ *
+ *  * Example usaging transiant storage:
```
Comment (Amxx) [inline scripts/generate/templates/StorageSlot.js:34]:
```suggestion
 *  * Example using transient storage:
```

**Review by ernestognw — DISMISSED** [review]
Just added a quick example in the utilities page.

LGTM

**Review by ernestognw — DISMISSED** [review]

**Review by ernestognw — APPROVED** [review]

**Pull request merged**